network/discovery/dv5_service.go

```diff
@@ -196,6 +219,36 @@ func (dvs *DiscV5Service) checkPeer(logger *zap.Logger, e PeerEvent) error {
 		return errors.New("no shared subnets")
 	}
 
+	if dvs.withExtraFiltering {
+		helpfulPeer := false // whether this peer helps us with getting rid of dead/solo subnets
+		subscribedTopics := dvs.topicsCtrl.Topics()
+		for _, topic := range subscribedTopics {
+			topicPeers, err := dvs.topicsCtrl.Peers(topic)
+			if err != nil {
+				return errors.Wrap(err, "could not get subscribed topic peers")
+			}
+
+			if len(topicPeers) >= 2 {
+				continue // this topic has enough peers
+			}
```

Comment:
This means we are looking (in discovery) for peers who help with either **dead** or **solo** subnets (without actually prioritizing the dead ones, and also this kind of filtering is maybe too aggressive overall since in my testing I see that we aren't keeping at `MaxPeers` limit for long - and rather falling 10-20% below it and keeping at that level)

this isn't optimal way to do it, but in my testing on prod I found it to be working reasonably good (with 60 and 90`MaxPeers`) - so we **probably want to improve this, just maybe not in the upcoming release**